test/authentication/authentication.test.js

```diff
@@ -267,135 +271,175 @@ describe('auth0.authentication', function() {
-      done
-    ) {
-      this.auth0.getSSOData(function(err, result) {
-        expect(err).to.be(null);
-        expect(result).to.be.eql({
-          lastUsedConnection: { name: 'lastUsedConnection' },
-          lastUsedUserID: 'the-user-id',
-          lastUsedUsername: 'last-used-user-name',
-          lastUsedClientID: '...',
-          sessionClients: ['...'],
-          sso: true
+        this.webAuthSpy.checkSession.lastCall.args[1](null, {
+          idTokenPayload: { sub: 'the-user-id', name: 'last-used-user-name' }
         });
-        done();
       });
+      it('returns ssoData object with lastUsedConnection and idtokenpayload.email by default', function(
+        done
+      ) {
+        this.auth0.getSSOData(function(err, result) {
+          expect(err).to.be(null);
+          expect(result).to.be.eql({
+            lastUsedConnection: { name: 'lastUsedConnection' },
+            lastUsedUserID: 'the-user-id',
+            lastUsedUsername: 'last-used-user-email',
+            lastUsedClientID: '...',
+            sessionClients: ['...'],
+            sso: true
+          });
+          done();
+        });
 
-      this.webAuthSpy.checkSession.lastCall.args[1](null, {
-        idTokenPayload: { sub: 'the-user-id', name: 'last-used-user-name' }
+        this.webAuthSpy.checkSession.lastCall.args[1](null, {
+          idTokenPayload: {
+            sub: 'the-user-id',
+            email: 'last-used-user-email',
+            name: 'do not use me'
+          }
+        });
       });
     });
-    it('returns ssoData object with lastUsedConnection and idtokenpayload.email by default', function(
-      done
-    ) {
-      this.auth0.getSSOData(function(err, result) {
-        expect(err).to.be(null);
-        expect(result).to.be.eql({
-          lastUsedConnection: { name: 'lastUsedConnection' },
-          lastUsedUserID: 'the-user-id',
-          lastUsedUsername: 'last-used-user-email',
-          lastUsedClientID: '...',
-          sessionClients: ['...'],
-          sso: true
+    context('when inside of the hosted login page', function() {
+      before(function() {
+        this.auth0 = new Authentication(this.webAuthSpy, {
+          domain: 'me.auth0.com',
+          clientID: '...',
+          redirectUri: 'http://page.com/callback',
+          responseType: 'code',
+          _sendTelemetry: false
         });
-        done();
       });
-
-      this.webAuthSpy.checkSession.lastCall.args[1](null, {
-        idTokenPayload: { sub: 'the-user-id', email: 'last-used-user-email', name: 'do not use me' }
+      beforeEach(function() {
+        stub(windowHelper, 'getWindow', function() {
+          return { location: { host: 'me.auth0.com' } };
+        });
+      });
+      afterEach(function() {
+        windowHelper.getWindow.restore();
+      });
+      it('calls webauth._universalLogin.getSSOData with same params', function() {
```

Comment:
Please explain me this test case. Why is it handling the same parameters to the other class's method and why is it accepting strings when it should be a boolean and a function?

Reply:
in the ULP, we need to run the exact same code we were using in v8, so I'm just calling the old getSSOData (now inside the HostedPages class) with the same params. I don't really care what are the params now, just want to test that I'm just relaying the params correctly